src/main/java/de/rabea/controller/MethodOptionsController.java

```diff
@@ -1,43 +1,44 @@
 package de.rabea.controller;
 
+import de.rabea.ContentStorage;
 import de.rabea.Controller;
 import de.rabea.request.HttpRequest;
 import de.rabea.request.HttpVerb;
-import de.rabea.response.HttpResponse;
-import de.rabea.response.OptionsResponseHeader;
+import de.rabea.response.*;
+import de.rabea.response.creator.*;
+import de.rabea.response.head.OptionsResponseHeader;
 
 import java.util.HashMap;
 import java.util.Map;
 
 import static de.rabea.request.HttpVerb.*;
-import static de.rabea.response.StatusLine.OK;
+import static de.rabea.response.head.StatusLine.NOT_ALLOWED;
+import static de.rabea.response.head.StatusLine.OK;
 
-public class MethodOptionsController extends Controller {
-    private Map<HttpVerb, HttpResponse> responsesForMethods = new HashMap<>();
-    private HttpRequest request;
+public class MethodOptionsController implements Controller {
+    private Map<HttpVerb, ResponseCreator> responsesForMethods = new HashMap<>();
+    private ContentStorage contentStorage;
 
-    public MethodOptionsController() {
+    public MethodOptionsController(ContentStorage contentStorage) {
+        this.contentStorage = contentStorage;
         this.responsesForMethods = registerResponses();
-        this.request = null;
     }
 
     @Override
-    public HttpResponse getResponse(HttpRequest httpRequest) {
-        this.request = httpRequest;
-        return responsesForMethods.getOrDefault(request.requestLine().method(), methodNotAllowed());
+    public HttpResponse getResponse(HttpRequest request) {
+        return responsesForMethods.getOrDefault(
+                request.requestLine().method(),
+                new GetResponseCreator(NOT_ALLOWED))
```

Comment:
`new GetResponseCreator(NOT_ALLOWED))` seems to be a common default here... I see it at least three times. Plus, it is weird that a `GetResponseCreator` is used to respond when calling `PUT` on a route that does not support it.